### ISSUE TITLE
Only datasets can be used as format_source for tool output.

### ISSUE THREAD

**Issue by nsoranzo** [galaxyproject/galaxy#699]
Dataset collections or their elements cannot be used as `format_source` (or I'm doing something wrong).

Use case: quality control filtering of paired FASTQ files. Input is a paired dataset collection (format can be fastq, fastqsanger, fastqillumina or fastqsolexa). Outputs are:
- a paired dataset collection with the reads passing the quality filters
- a dataset contained the singletons, i.e. the reads for which only one of the pair passed the filters.

For both outputs, the format should be the same of the input collection.
For the output collection I can use:

``````
<collection name="output_paired_coll" type="paired" structured_like="input_paired" inherit_format="true" label="Paired-end output of ${tool.name} on ${on_string}" />```
``````

but for the singletons, this does not work (output format is `data`):

```
<data name="output_paired_coll_single" format_source="input_paired" label="Singletons from paired-end output of ${tool.name} on ${on_string}">
```

I am also not sure how the format of a dataset collection is defined.

**Comment by jmchilton** [comment]
The collection may have many different formats right, hmm... I could give you one at random of course but I'm not sure that is ideal. Would it be better to ask the user to select a format in a conditional and then use that to make the data selector more specific and use output actions to format this dataset?

**Comment by nsoranzo** [comment]
@jmchilton That's an option, but it would make it more complicated for the user when collection are used then when separate files are used (in which case the format of the forward FASTQ file is used for the outputs).

If `format_source` would accept `input_paired.forward` I would be happy :smiley: 

The full wrapper is [here](https://github.com/galaxyproject/tools-iuc/pull/369).

**Comment by bgruening** [comment]
I also like the idea to access `input_paired.forward`.

**Comment by nsoranzo** [comment]
Fixed by #1113 and #1155.
